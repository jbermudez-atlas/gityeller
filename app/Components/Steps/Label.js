import React from 'react';
import axios from 'axios';

export default class LabelStep extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      labels: [],
      labelChosen: ''
    };
  }
  stepLoaded() {
    return this.props.activeStep === 2;
  }
  componentDidUpdate(props) {
    if (this.stepLoaded() && props.activeStep !== this.props.activeStep) {
      this.setState({labels: []});

      axios.get(`/api/repo/${this.props.repo}/labels`)
        .then((res) => this.setState({labels: res.data}))
        .catch(() => this.previous())
    }
  }
  chooseLabel(e) {
    if (e.target.value) {
      this.props.next({
        labelChosen: e.target.value,
        activeStep: this.props.activeStep + 1
      })
    }
  }
  previous() {
    this.props.next({
      activeStep: this.props.activeStep - 1
    })
  }
  render() {
    return (
      <div className={this.stepLoaded() ? this.props.styles.activeStep : this.props.styles.inactiveStep}>
        <div className="form-group">
          <label htmlFor="labels">Labels</label>
          {this.state.labels.length === 0 &&
            <div>
              <br/>
              <b>Wait for it ...</b>
            </div>
          }
          {this.state.labels.length > 0 &&
            <select
              className={`form-control ${this.props.styles.customControl}`}
              id="labels"
              onChange={this.chooseLabel.bind(this)}
              autoComplete="off"
            >
              <option>Select a label</option>
              {this.state.labels.map((label, id) => {
                return <option id={label.id} key={id}>{label.name}</option>;
              })}
            </select>
          }
          <div className={`row ${this.props.styles.actions}`}>
            <div className="col-xs-6">
              {this.props.activeStep !== 1 &&
                <button
                  className={`btn btn-default ${this.props.styles.customDefaultButton}`}
                  onClick={this.props.startOver}>
                    Reset
                </button>
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

LabelStep.propTypes = {
  activeStep: React.PropTypes.number.isRequired,
  next: React.PropTypes.func.isRequired,
  styles: React.PropTypes.object.isRequired,
  repo: React.PropTypes.string,
  startOver: React.PropTypes.func.isRequired
};
