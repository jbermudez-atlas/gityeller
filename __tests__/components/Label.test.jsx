import React from 'react';
import Label from './../../app/Components/Steps/Label';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('Take Snapshot', () => {
  const component = renderer.create(
    <Label
      activeStep={2}
      repo="kbariotis/ansible-nodejs-digitalocean"
      next={function(){}}
      startOver={function(){}}
      styles={{}}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
