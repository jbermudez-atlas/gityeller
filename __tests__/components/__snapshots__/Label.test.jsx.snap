exports[`test Take Snapshot 1`] = `
<div
  className={undefined}>
  <div
    className="form-group">
    <label
      htmlFor="labels">
      Labels
    </label>
    <div>
      <br />
      <b>
        Wait for it ...
      </b>
    </div>
    <div
      className="row undefined">
      <div
        className="col-xs-6">
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          Reset
        </button>
      </div>
    </div>
  </div>
</div>
`;
